### PR TITLE
Always define request-permissions

### LINKED ISSUE #5571: status_im.utils.notifications.request_permissions is not a function after creating account and first log in
### Description

*Type*: Bug

*Summary*: after creating account and first log in I'm getting an error `status_im.utils.notifications.request_permissions  is not a function`  and when I click on "OK', I can proceed.

#### Expected behavior
no error is shown

### Reproduction

- Open Status
- Create an account
- Log in

### Additional Information
* Status version: nightly 21/07/18
* Operating System: Mac OSx

#### Logs

[StatusIm.txt](https://github.com/status-im/status-react/files/2305361/StatusIm.txt)

### PR DESCRIPTION
**Pull request by vitvly**
Fixes #5571

### CODE REVIEW

File: src/status_im/notifications/core.cljs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (vkjr) [inline src/status_im/notifications/core.cljs]: empty line :)

**Review by vkjr — APPROVED** [review]

**Review by goranjovic — APPROVED** [review]

**Comment by goranjovic** [comment]
@siphiuel don't forget to squash before merging

File: src/status_im/notifications/core.cljs (hunk 1 above)
Comment (vitvly) [reply src/status_im/notifications/core.cljs]: Sorry, fixed.

**Comment by vitvly** [comment]
@goranjovic Done

File: src/status_im/notifications/core.cljs [hunk 2]
```diff
@@ -13,20 +13,21 @@
...
+    (re-frame/dispatch [:notifications/request-notifications-granted {}])
+    (-> (.requestPermission (.messaging firebase))
+        (.then
+         (fn [_]
+           (log/debug "notifications-granted")
+           (re-frame/dispatch [:notifications/request-notifications-granted {}]))
+         (fn [_]
+           (log/debug "notifications-denied")
+           (re-frame/dispatch [:notifications/request-notifications-denied {}]))))))
+(when-not platform/desktop?
```
Comment (yenda) [inline src/status_im/notifications/core.cljs]: Can you add an empty line here to make it clear that it is not part of the function above ?

**Review by yenda — APPROVED** [review]

File: src/status_im/notifications/core.cljs (hunk 2 above)
Comment (vitvly) [reply src/status_im/notifications/core.cljs]: Done.

**Comment by churik** [comment]
Cannot create new account, hangs on "Creating your account" screen.
```
Debug: starting ubuntu server...
 
Debug: wait for started...
 
Debug: ubuntu-server std:  "outerRealmConstructor: undefined"
 
Debug: ubuntu-server std:  "Loaded realmConstructor: function Realm() { [native code] }"
 
Debug: ubuntu-server err:  "-- Server starting"
 
Debug: waiting finished
 
Debug: ReactNativeConfig Can't open config file  "/Applications/StatusIm.app/Contents/MacOS/../../.env"  Error code:  5
 
Warning: WebEngineContext used before QtWebEngine::initialize() or OpenGL context creation failed.
 
Debug: ubuntu-server err:  "-- Connection from RN client"
 
Debug: ubuntu-server std:  "outerRealmConstructor: function"
 
Debug: ubuntu-server std:  "Running application \"StatusIm\" with appParams: {\"initialProps\":{},\"rootTag\":1}. __DEV__ === false, development-level warning are OFF, performance optimizations are ON"
 
Debug: invoked RNKeychainManager::setUsername with username =  "status-im.encryptionkey"
 
Debug: invoked RNKeychainManager::getGenericPasswordForOptions
 
Debug: RNKeychainManager::getGenericPasswordForOptions success
 
Debug: ubuntu-server err:  "re-frame: no handler registered for effect: { ns: 'notifications',\n  name: 'handle-initial-push-notification',\n  fqn: 'notifications/handle-initial-push-notification',\n  _hash: 1143053946,\n  'cljs$lang$protocol_mask$partition0$': 2153775105,\n  'cljs$lang$protocol_mask$partition1$': 4096 } . Ignoring."
 
Debug: call of RCTStatus::getDeviceUUID
 
Debug: ubuntu-server err:  "ERROR [status-im.data-store.core:25] - Could not move realms TypeError: re-matches must match against a string."
 
Debug: call of RCTStatus::startNode with param configString: "null"
 
Debug: "illegal value"
 
Debug:  RCTStatus::startNode configString:  QMap()
 
Debug: RCTStatus::startNode networkDir:  "/Users/curikovatm/Library/Application Support/StatusIm/"
 
Debug: RCTStatus::startNode GenerateConfig result:  {"NetworkId":0,"DataDir":"/Users/curikovatm/Library/Application Support/StatusIm/","KeyStoreDir":"/Users/curikovatm/Library/Application Support/StatusIm/keystore","NodeKeyFile":"","NoDiscovery":false,"Rendezvous":false,"ListenAddr":":0","AdvertiseAddr":"","Name":"StatusIM","Version":"0.9.9-","APIModules":"eth,net,web3,peer","HTTPHost":"localhost","RPCEnabled":false,"HTTPPort":8545,"IPCFile":"geth.ipc","IPCEnabled":false,"TLSEnabled":false,"MaxPeers":25,"MaxPendingPeers":0,"LogEnabled":false,"LogFile":"","LogLevel":"ERROR","LogToStderr":true,"UpstreamConfig":{"Enabled":false,"URL":""},"ClusterConfigFile":"","ClusterConfig":{"Enabled":true,"Fleet":"eth.beta","StaticNodes":[],"BootNodes":[],"TrustedMailServers":null,"RendezvousNodes":null},"LightEthConfig":{"Enabled":true,"Genesis":"","DatabaseCache":16},"WhisperConfig":{"Enabled":true,"LightClient":false,"EnableMailServer":false,"DataDir":"/Users/curikovatm/Library/Application Support/StatusIm/wnode","MinimumPoW":0.001,"MailServerPasswordFile":"","MailServerPassword":"","MailServerAsymKeyFile":"","MailServerAsymKey":null,"MailServerRateLimit":0,"MailServerCleanupPeriod":0,"TTL":120,"FirebaseConfig":{"AuthorizationKeyFile":"","NotificationTriggerURL":"https://fcm.googleapis.com/fcm/send"},"EnableNTPSync":true},"SwarmConfig":{"Enabled":false},"RegisterTopics":[],"RequireTopics":{"whisper":{"Min":2,"Max":2}},"StatusServiceEnabled":false,"DebugAPIEnabled":false,"MailServerRegistryAddress":""}
 
Debug:  RCTStatus::startNode GenerateConfig configString:  QMap(("APIModules", QVariant(QString, "eth,net,web3,peer"))("AdvertiseAddr", QVariant(QString, ""))("ClusterConfig", QVariant(QVariantMap, QMap(("BootNodes", QVariant(QVariantList, ()))("Enabled", QVariant(bool, true))("Fleet", QVariant(QString, "eth.beta"))("RendezvousNodes", QVariant(std::nullptr_t, (nullptr)))("StaticNodes", QVariant(QVariantList, ()))("TrustedMailServers", QVariant(std::nullptr_t, (nullptr))))))("ClusterConfigFile", QVariant(QString, ""))("DataDir", QVariant(QString, "/Users/curikovatm/Library/Application Support/StatusIm/"))("DebugAPIEnabled", QVariant(bool, false))("HTTPHost", QVariant(QString, "localhost"))("HTTPPort", QVariant(double, 8545))("IPCEnabled", QVariant(bool, false))("IPCFile", QVariant(QString, "geth.ipc"))("KeyStoreDir", QVariant(QString, "/Users/curikovatm/Library/Application Support/StatusIm/keystore"))("LightEthConfig", QVariant(QVariantMap, QMap(("DatabaseCache", QVariant(double, 16))("Enabled", QVariant(bool, true))("Genesis", QVariant(QString, "")))))("ListenAddr", QVariant(QString, ":0"))("LogEnabled", QVariant(bool, false))("LogFile", QVariant(QString, ""))("LogLevel", QVariant(QString, "ERROR"))("LogToStderr", QVariant(bool, true))("MailServerRegistryAddress", QVariant(QString, ""))("MaxPeers", QVariant(double, 25))("MaxPendingPeers", QVariant(double, 0))("Name", QVariant(QString, "StatusIM"))("NetworkId", QVariant(double, 0))("NoDiscovery", QVariant(bool, false))("NodeKeyFile", QVariant(QString, ""))("RPCEnabled", QVariant(bool, false))("RegisterTopics", QVariant(QVariantList, ()))("Rendezvous", QVariant(bool, false))("RequireTopics", QVariant(QVariantMap, QMap(("whisper", QVariant(QVariantMap, QMap(("Max", QVariant(double, 2))("Min", QVariant(double, 2))))))))("StatusServiceEnabled", QVariant(bool, false))("SwarmConfig", QVariant(QVariantMap, QMap(("Enabled", QVariant(bool, false)))))("TLSEnabled", QVariant(bool, false))("UpstreamConfig", QVariant(QVariantMap, QMap(("Enabled", QVariant(bool, false))("URL", QVariant(QString, "")))))("Version", QVariant(QString, "0.9.9-"))("WhisperConfig", QVariant(QVariantMap, QMap(("DataDir", QVariant(QString, "/Users/curikovatm/Library/Application Support/StatusIm/wnode"))("EnableMailServer", QVariant(bool, false))("EnableNTPSync", QVariant(bool, true))("Enabled", QVariant(bool, true))("FirebaseConfig", QVariant(QVariantMap, QMap(("AuthorizationKeyFile", QVariant(QString, ""))("NotificationTriggerURL", QVariant(QString, "https://fcm.googleapis.com/fcm/send")))))("LightClient", QVariant(bool, false))("MailServerAsymKey", QVariant(std::nullptr_t, (nullptr)))("MailServerAsymKeyFile", QVariant(QString, ""))("MailServerCleanupPeriod", QVariant(double, 0))("MailServerPassword", QVariant(QString, ""))("MailServerPasswordFile", QVariant(QString, ""))("MailServerRateLimit", QVariant(double, 0))("MinimumPoW", QVariant(double, 0.001))("TTL", QVariant(double, 120))))))
 
Debug: RCTStatus::startNode StartNode result:  {"error":"Key: 'NodeConfig.NetworkID' Error:Field validation for 'NetworkID' failed on the 'required' tag"}
 
Warning: QCoreApplication::postEvent: Unexpected null receiver
 
Warning: QCoreApplication::postEvent: Unexpected null receiver
 
Warning: QCoreApplication::postEvent: Unexpected null receiver
 
Warning: QCoreApplication::postEvent: Unexpected null receiver
 
Warning: QVariantMap (anonymous namespace)::makeReactTouchEvent(QQuickItem *, QMouseEvent *) target was not a reactItem
 
Warning: QVariantMap (anonymous namespace)::makeReactTouchEvent(QQuickItem *, QMouseEvent *) target was not a reactItem
 
Warning: QVariantMap (anonymous namespace)::makeReactTouchEvent(QQuickItem *, QMouseEvent *) target was not a reactItem
 
Warning: QCoreApplication::postEvent: Unexpected null receiver
 
Warning: QVariantMap (anonymous namespace)::makeReactTouchEvent(QQuickItem *, QMouseEvent *) target was not a reactItem
 
Warning: QVariantMap (anonymous namespace)::makeReactTouchEvent(QQuickItem *, QMouseEvent *) target was not a reactItem
 
Warning: QVariantMap (anonymous namespace)::makeReactTouchEvent(QQuickItem *, QMouseEvent *) target was not a reactItem
 
Debug: call of RCTStatus::createAccount with param callbackId:  497
 
Debug: RCTStatus::createAccount CreateAccount result:  {"address":"","pubkey":"","mnemonic":"","error":"geth node is not available"}
 
Warning: QCoreApplication::postEvent: Unexpected null receiver

```

**Comment by vitvly** [comment]
I see a strange error `Debug: ubuntu-server err:  "ERROR [status-im.data-store.core:25] - Could not move realms TypeError: re-matches must match against a string."`

**Comment by vitvly** [comment]
Correction: key errors are
`Debug: RCTStatus::startNode StartNode result:  {"error":"Key: 'NodeConfig.NetworkID' Error:Field validation for 'NetworkID' failed on the 'required' tag"}`
and as a consequence
`Debug: RCTStatus::createAccount CreateAccount result:  {"address":"","pubkey":"","mnemonic":"","error":"geth node is not available"}`

**Review by maxhora — APPROVED** [review]

**Comment by churik** [comment]
@siphiuel
I still have the same issue.

**Comment by vitvly** [comment]
App hanging on `Create Account` screen was not specific to this screen and was caused by `.env` file missing from the root of app distribution. https://github.com/status-im/status-react/pull/5752 will fix this problem, after the latter is merged this PR can be retested.

**Comment by vitvly** [comment]
Rebased with latest develop.

**Comment by churik** [comment]
Tested (Desktop):
- creating account;
- recovering account;
- log in with account, created on previous nightly;
- fetching messages for 1-1, public chats;


Tested (IOS):
- creating account;
- recovering account;

**Pull request merged**